client/address.go

```diff
@@ -0,0 +1,41 @@
+package client
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+)
+
+func SanitizeAddress(address string) (string, error) {
+	u, err := url.Parse(address)
+	if err != nil {
+		return "", fmt.Errorf("error parsing peer URL: %w", err)
+	}
+
+	hostname, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return "", fmt.Errorf("error splitting host port from parsed URL: %w", err)
+	}
+
+	if strings.Contains(hostname, ":") {
+		// IPv6 Addreses need to be wrapped in brackets
+		return fmt.Sprintf("[%s]:%s", hostname, port), nil
+	} else {
+		return fmt.Sprintf("%s:%s", hostname, port), nil
+	}
+}
```

Comment:
This could be simplified with `net.JoinHostPort` which already handles ipv6 for you.

Reply:
I think instead of splitting then joining host and port, you can simply return `u.Host`. It already removes the scheme and `url.Parse` should already sanitize the address for you.